site/_posts/2020-07-02-auth-and-the-bottleneck-architecture.md

```diff
@@ -0,0 +1,83 @@
+---
+layout: blog-post
+title: The Auth Bottleneck Pattern 
+description: How and why to use a centralized user management system.
+author: Dan Moore
+image: blogs/private-label-with-tenants/how-to-use-fusionauths-multi-tenant-feature-to-create-a-private-label-offering.png
+category: blog
+excerpt_separator: "<!--more-->"
+---
+
+One common pattern for modern organizations is to centralize user management with a bottleneck architecture. A solid user management system is provisioned and all authentication and authorization requests are routed through it, rather than individual applications having their own auth components. 
+
+<!--more-->
+
+## The Pattern
+
+In this architectural pattern, applications delegate user management to the user management system. Many kinds of applications can perform such delegation. Examples include custom applications written by internal teams, third party services used by employees, such as Salesforce and Zendesk, and applications used by customers, such as forums or account management. 
+
+Like any serious architecture discussion, we'll need some acronyms and jargon. There are a few identity management standards, including OpenID Connect (OIDC), released in the mid 2010s, and Security Assertion Markup Language (SAML), which was standardized in the 2000s.
+
+If the protocol in use is OIDC, applications delegating auth decisions are called Relying Parties, or RPs. If the protocol is SAML, the delegating applications are called Service Providers, or SPs. In either case, decisions about authentication or authorization, and often account management, are shifted from the application to a central service.
+
+In OIDC systems, this centralized auth system is referred to as an OpenID provider, or OP. For SAML based systems, it is called an Identity Provider, or IdP. The system's scope includes:
+
+* authentication - who is this user? 
+* authorization - what can this user do?
+* user management - registration, forgot password flows, two factor authentication, reporting, and more
+
+However, an auth system isn't limited to just verifying users locally. It can also federate to other providers of identity, such as Google, HYPR or Facebook. In addition, standards compliant providers adhering to the SAML or OIDC specifications can be integrated.
+
+Here's a diagram of the bottleneck architecture:
+
+{% include _image.liquid src="/assets/img/blogs/bottleneck-pattern/bottleneck-diagram.png" alt="A common archictural diagram." class="img-fluid" figure=false %}
```

Comment:
Total nit, but can we make the FusionAuth box our dark grey color and use the white icon with orange dots? :)

Reply:
Donesky.